### PR TITLE
WIP: Tzip 10 wallet provider api

### PR DESCRIPTION
**Pull request by carte7000**

### CODE REVIEW

File: packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  private removeFeeAndLimit<T extends { gas_limit: any, storage_limit: any, fee: any }>(op: T) {
+    const { fee, gas_limit, storage_limit, ...rest } = op;
+    return rest;
+  }
+
+  async originate(contract: WalletOriginateParams): Promise<OriginationWalletOperation> {
+    const op = await createOriginationOperation(contract as any)
+    const network = this.permissions!.permissions.networks;
```
Comment (jevonearth) [inline packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts]: We will need to handle the permissions here.

File: packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  message = 'You need to initialize BeaconWallet by calling beaconWallet.requestPermissions first';
+}
+
+export class BeaconWallet implements WalletProvider {
+  public client: DAppClient;
+
+  private permissions?: PermissionResponse;
+
+  constructor(name: string) {
+    this.client = new DAppClient(name);
```
Comment (carte7000) [inline packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts]: We need to provide more option to configure the dApp client. Probably allowing the user to pass in the DAppClient instance would be better.

Comment (AndreasGassmann) [reply packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts]: Yes, there will probably be more options to configure the dApp client soon (eg. disabling standard behaviour like showing the QR code alert).

What if we introduced a `DAppClientOptions` object that can be passed in the constructor, so you can use the same type in your constructor?

Comment (carte7000) [reply packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts]: Good idea, let's do it that way :+1:

**Pull request merged**